### PR TITLE
Remove capabilities on plugin uninstall

### LINKED ISSUE #3827: Remove capabilities on plugin uninstall
## Feature Description

When the plugin is uninstalled, the custom capabilities added by the plugin should get removed as well.

Introduced in #3507.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
When the plugin is uninstalled, the custom capabilities added by the plugin should get removed as well.

Introduced in #3507.

## Relevant Technical Choices
Decided to keep the logic inside the story post type class. 

## Testing Instructions

---

Fixes #3827

### CODE REVIEW

File: includes/Story_Post_Type.php [hunk 1]
```diff
@@ -243,6 +243,38 @@ public function add_caps_to_roles() {
...
+	public function remove_caps_to_roles() {
+		$post_type_object = get_post_type_object( self::POST_TYPE_SLUG );
+
+		if ( ! $post_type_object ) {
+			return;
+		}
+
+		$all_capabilities = array_values( (array) $post_type_object->cap );
+		$roles            = wp_roles();
+		foreach ( $roles->role_objects as $name => $role ) {
```
Comment (swissspidy) [inline includes/Story_Post_Type.php]:
```suggestion
		foreach ( $roles->role_objects as $role ) {
```

File: includes/Story_Post_Type.php [hunk 2]
```diff
@@ -243,6 +243,38 @@ public function add_caps_to_roles() {
...
+				foreach ( $all_capabilities as $cap ) {
+					$role->remove_cap( $cap );
+				}
+			}
+		}
+
+		/**
+		 * Fires when removing the custom capabilities to existing roles.
+		 *
+		 * Can be used to remove the capabilities to other, custom roles.
```
Comment (swissspidy) [inline includes/Story_Post_Type.php]:
```suggestion
		 * Fires when removing the custom capabilities from existing roles.
		 *
		 * Can be used to remove the capabilities from other, custom roles.
```

File: tests/phpunit/tests/Story_Post_Type.php [hunk 3]
```diff
@@ -302,6 +302,27 @@ public function test_add_caps_to_roles() {
 		}
 	}
 
+
+	/**
+	 * @covers ::remove_caps_to_roles
+	 */
+	public function test_remove_caps_to_roles() {
+		$story_post_type = new \Google\Web_Stories\Story_Post_Type( $this->createMock( \Google\Web_Stories\Experiments::class ) ); - +
```
Comment (swissspidy) [inline tests/phpunit/tests/Story_Post_Type.php]:
```suggestion
		$story_post_type = new \Google\Web_Stories\Story_Post_Type( $this->createMock( \Google\Web_Stories\Experiments::class ) );
```

File: includes/Story_Post_Type.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php]: Fixed

File: includes/Story_Post_Type.php [hunk 4]
```diff
@@ -250,6 +250,39 @@ public function add_caps_to_roles() {
 		do_action( 'web_stories_add_capabilities', $all_capabilities );
 	}
 
+	/**
+	 * Removes story capabilities to all user roles.
+	 *
+	 * @return void
+	 */
+	public function remove_caps_to_roles() {
```
Comment (swissspidy) [inline includes/Story_Post_Type.php]:
```suggestion
	/**
	 * Removes story capabilities from all user roles.
	 *
	 * @return void
	 */
	public function remove_caps_from_roles() {
```

File: includes/uninstall.php [hunk 5]
```diff
@@ -130,6 +130,15 @@ function delete_posts() {
...
 	}
 }
+/**
+ * Remove user capabilities.
+ *
+ * @return void
+ */
+function remove_caps() {
+	$story_post_type = new Story_Post_Type( new Experiments() );
+	$story_post_type->remove_caps_to_roles();
```
Comment (swissspidy) [inline includes/uninstall.php]:
```suggestion
	$story_post_type->remove_caps_from_roles();
```

File: tests/phpunit/tests/Story_Post_Type.php [hunk 6]
```diff
@@ -304,6 +304,27 @@ public function test_add_caps_to_roles() {
 		}
 	}
 
+
+	/**
+	 * @covers ::remove_caps_to_roles
+	 */
+	public function test_remove_caps_to_roles() {
+		$story_post_type = new \Google\Web_Stories\Story_Post_Type( $this->createMock( \Google\Web_Stories\Experiments::class ) );
+		$story_post_type->remove_caps_to_roles();
```
Comment (swissspidy) [inline tests/phpunit/tests/Story_Post_Type.php]:
```suggestion
	/**
	 * @covers ::remove_caps_from_roles
	 */
	public function test_remove_caps_from_roles() {
		$story_post_type = new \Google\Web_Stories\Story_Post_Type( $this->createMock( \Google\Web_Stories\Experiments::class ) );
		$story_post_type->remove_caps_from_roles();
```

**Review by swissspidy — APPROVED** [review]

**Pull request merged**